firmware/hw_layer/stepper_dual_hbridge.cpp

```diff
@@ -36,19 +50,57 @@ bool DualHBridgeStepper::step(bool positive) {
 		return false;
 	}
 
+	if (CONFIG(stepperNumMicroSteps) > 1) {
+		float dutyMult = CONFIG(stepperMaxDutyCycle) * phaseDutyCycleDivisor;
+		int numStepIncr = maxNumSteps / CONFIG(stepperNumMicroSteps);
+		if (!positive)
+			numStepIncr = -numStepIncr;
+		for (int i = CONFIG(stepperNumMicroSteps); i > 0; i--) {
+			m_phase = (m_phase + numStepIncr) & tableSizeMask;
+			update(dutyMult);
+			// sleep 1/Nth of the pause time
+			pause(CONFIG(stepperNumMicroSteps));
+		}
+		return true;
+	}
+
     // step phase, wrapping
     if (positive) {
```

Comment:
why special case non-micro stepping? can't we just use the same phase table even for full stepping? I think all the math works out properly for 1x stepping.

Reply:
Sure we can, but the simple and straightforward approach with just "numSteps=1" will give us a so-called "wave drive" method (aka "one phase on"):
![image](https://user-images.githubusercontent.com/15104768/131781721-4d48abd4-27ac-4c9c-9135-e9834f70a2b6.png)
And unfortunately it would give less torque than the current "full step max torque" method (aka "two phase on"):
![image](https://user-images.githubusercontent.com/15104768/131781778-207098e4-873c-4a32-a4a0-d751ee31f9b9.png)

Max torque drive on the left and wave drive on the right:
![image](https://user-images.githubusercontent.com/15104768/131784497-04c755f2-90bd-4a6a-b773-3c3d0641d9cb.png)

However it's ok to use the wave method for half-step (and microstepping): the torque loss is not that substantial because of corner smoothing:

![image](https://user-images.githubusercontent.com/15104768/131784664-7588de51-8597-41b6-a069-e242f1be40dd.png)

![image](https://user-images.githubusercontent.com/15104768/131784706-5dd7e1a7-8e1f-4784-abf8-c13de1df53c3.png)

Reply:
Ah hah, makes sense.  Mention that in a comment in-code!